pyvista/plotting/composite_mapper.py

```diff
@@ -0,0 +1,769 @@
+
+        Examples
+        --------
+        Change the opacity of the second block of the dataset then reset all
+        opacities.
+
+        Note how the zero index is the entire multiblock, so we have to add 1
+        to our indexing to access the right block.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.block_attr[2].opacity = 0.5
+        >>> mapper.block_attr.reset_opacities()
+        >>> pl.show()
+
+        """
+        self.RemoveBlockOpacities()
+
+    def get_block(self, index):
+        """Return a block by its flat index.
+
+        Parameters
+        ----------
+        index : int
+            Flat index of the block to retrieve.
+
+        Returns
+        -------
+        pyvista.DataObject
+            PyVista dataobject.
+
+        Examples
+        --------
+        Note how the zero index is the entire multiblock. To access individual
+        multiblocks, add one to the index.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.block_attr.get_block(0)  # doctest:+SKIP
+        MultiBlock (...)
+          N Blocks:	2
+          X Bounds:	-0.500, 0.500
+          Y Bounds:	-0.500, 0.500
+          Z Bounds:	-0.500, 1.500
+
+        Note this is the same as using ``__getitem__``
+
+        >>> mapper.block_attr[0]  # doctest:+SKIP
+        MultiBlock (...)
+          N Blocks:	2
+          X Bounds:	-0.500, 0.500
+          Y Bounds:	-0.500, 0.500
+          Z Bounds:	-0.500, 1.500
+
+        """
+        try:
+            if vtk_version_info <= (9, 0, 3):  # pragma: no cover
+                vtk_ref = _vtk.reference(0)  # needed for <=9.0.3
+                block = self.DataObjectFromIndex(index, self._dataset, vtk_ref)
+            else:
+                block = self.DataObjectFromIndex(index, self._dataset)
+        except OverflowError:
+            raise KeyError(f'Invalid block key: {index}') from None
+        if block is None:
+            if index > len(self):
+                raise KeyError(
+                    f'index {index} is out of bounds. There are only {len(self)} blocks.'
+                ) from None
+        return block
+
+    def __getitem__(self, index):
+        """Return a block attribute by its flat index."""
+        return BlockAttributes(self.get_block(index), self)
+
+    def __len__(self):
+        """Return the number of blocks."""
+        return len(self._dataset)
+
+    def __iter__(self):
+        """Return an iterator of all the block attributes."""
+        for ii in range(len(self)):
+            yield self[ii]
+
+
+class CompositePolyDataMapper(_vtk.vtkCompositePolyDataMapper2):
+    """Composite PolyData mapper.
```

Comment:
Currently this class shows up in the docs, but most of the methods only have a single-sentence docstring, no parameters/returns/examples. If this is final then we should open an issue to get these fixed (potentially as good first issues).

Reply:
Done in aee6ffa58ee0f26d183b464f761ef056f110cae2.